pkg/operator/operands/prometheus/resources.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2025 NVIDIA CORPORATION
+// SPDX-License-Identifier: Apache-2.0
+
+package prometheus
+
+import (
+	"context"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	kaiv1 "github.com/NVIDIA/KAI-scheduler/pkg/apis/kai/v1"
+)
+
+const (
+	mainResourceName = "kai"
+)
+
+func prometheusForKAIConfig(
+	ctx context.Context, runtimeClient client.Reader, kaiConfig *kaiv1.Config,
+) ([]client.Object, error) {
+	logger := log.FromContext(ctx)
+	config := kaiConfig.Spec.Prometheus
+
+	// Check if Prometheus is enabled
+	if config == nil || config.Enabled == nil || !*config.Enabled {
+		logger.Info("Prometheus is disabled in configuration")
+		return []client.Object{}, nil
+	}
+
+	logger.Info("Prometheus is enabled, checking for Prometheus Operator installation")
+
+	// Check if Prometheus Operator is installed by looking for the Prometheus CRD
+	// This is a simple check - in production you might want to check for the operator deployment
+	hasPrometheusOperator, err := CheckPrometheusOperatorInstalled(ctx, runtimeClient)
+	if err != nil {
+		logger.Error(err, "Failed to check for Prometheus Operator installation")
+		return nil, err
+	}
+
+	// If Prometheus Operator is not installed, we can't create a Prometheus CR
+	if !hasPrometheusOperator {
+		logger.Info("Prometheus Operator not found - Prometheus CRD is not available")
+		return []client.Object{}, nil
+	}
+
+	logger.Info("Prometheus Operator found, proceeding with Prometheus instance creation")
```

Comment:
This might be confusing, since this log will be printed every reconcile loop, but we will only create the prometheus once (if everything works right)
You're not "creating" you're "reconciling" (trying to move the prometheus to the desired state)